run-tests.sh

```diff
@@ -0,0 +1,28 @@
+#! /bin/bash -ex
+
+srcdir=$(dirname "$(readlink -f "$0")")
+cd "$srcdir"
+filename=$(basename "$(spectool -S vcs-diff-lint.spec | sed -n 's/Source1: \(.*\)/\1/p')")
+echo "$filename"
+rm -rf test-repo
+git clone "$filename" test-repo
+export PATH=$srcdir:$PATH
+cd test-repo
+
+for test_dir in "$srcdir"/test_data/*; do
+    test -d "$test_dir" || continue
+    test_name=$(basename "$test_dir")
+
+    echo "=== Test $test_name ==="
+    new=$(echo "$test_name" | cut -d_ -f2)
+    old=$(echo "$test_name" | cut -d_ -f3)
+    testdir=$srcdir/test_data/$test_name
+    git checkout "$new"
+    vcs-diff-lint --compare-against "$old" > errors || :
+
+    found_match=false
+    for added in "$testdir"/added-*.err; do
+        diff -ruN errors "$added" && found_match=: && break
+    done
+    $found_match
+done
```

Comment:
I'd rather see the tests implemented using pytest, unittest, or some other python test framework. But until we get to that point, this is good enough :-)

I am just expressing my preference for a proper rewrite before we start growing this to 100+ lines.

Reply:
Thank you for taking a look!  I ... I agree.

Do you think we can use pytest for this?  I.e. running vcs-diff-lint binary as-is?

Reply:
I've been playing with the idea of rewriting Copr beaker tests in Python (someday in the future, of course). So I tried many times to find some Python library that people use for writing their end-to-end or integration tests and I found nothing. Which leads me to a conclusion that either nobody does it (which doesn't make any sense to me) or everybody simply uses `pytest`.

I can very well imagine `pytest` tests heavily using `subprocess.run` for testing the binary itself.

Btw Tito does this as well - https://github.com/rpm-software-management/tito/blob/main/test/functional/multiproject_tests.py

Reply:
E.g. [behave](https://github.com/rpm-software-management/mock/pull/1480)?

Reply:
To be honest, I don't like `behave` because its whole premise is IMHO flawed. And if not flawed, at least not applicable to the majority of projects I came across. I am glad we tested it on our projects because it is a cool gimmick but personally, I wouldn't introduce behave tests to any other project.

I can elaborate more if you are interested.

Reply:
The more I use behave, the more I like it, so we can do some f2f chat about it..  Behave though is not a unit-testing framework for sure.  The case we have here is a task for pytest, indeed.